packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts

```diff
@@ -180,46 +180,58 @@ export class TestRunnerService {
 
 		// 2. Run over all the test cases
 
-		await this.testRunRepository.markAsRunning(testRun.id);
+		await this.testRunRepository.markAsRunning(testRun.id, pastExecutions.length);
 
 		// Object to collect the results of the evaluation workflow executions
 		const metrics = new EvaluationMetrics(testMetricNames);
 
 		for (const { id: pastExecutionId } of pastExecutions) {
-			// Fetch past execution with data
-			const pastExecution = await this.executionRepository.findOne({
-				where: { id: pastExecutionId },
-				relations: ['executionData', 'metadata'],
-			});
-			assert(pastExecution, 'Execution not found');
-
-			const executionData = parse(pastExecution.executionData.data) as IRunExecutionData;
-
-			// Run the test case and wait for it to finish
-			const testCaseExecution = await this.runTestCase(workflow, executionData, user.id);
-
-			// In case of a permission check issue, the test case execution will be undefined.
-			// Skip them and continue with the next test case
-			if (!testCaseExecution) {
-				continue;
+			try {
+				// Fetch past execution with data
+				const pastExecution = await this.executionRepository.findOne({
+					where: { id: pastExecutionId },
+					relations: ['executionData', 'metadata'],
+				});
+				assert(pastExecution, 'Execution not found');
+
+				const executionData = parse(pastExecution.executionData.data) as IRunExecutionData;
+
+				// Run the test case and wait for it to finish
+				const testCaseExecution = await this.runTestCase(workflow, executionData, user.id);
```

Comment:
What are your thoughts on not incrementing `failed` when there's a failure with the `testCaseExecution`? Do you think this should just be a concern for the evaluation workflow, or should this be also considered a failure?

Reply:
My assumption was that this should be evaluation workflow's concern. This will also allow to create tests for  unhappy paths.